tools/kagami/src/main.rs

```diff
@@ -774,3 +777,114 @@ mod tokens {
         }
     }
 }
+
+mod examples {
+    use clap::{Parser, Subcommand};
+    use color_eyre::Result;
+    use iroha_crypto::{KeyPair, PublicKey};
+
+    use super::*;
+
+    #[derive(Parser, Debug, Clone)]
+    pub struct Args {
+        #[clap(subcommand)]
+        usecase: Usecase,
+    }
+
+    impl<T: Write> RunArgs<T> for Args {
+        fn run(self, writer: &mut BufWriter<T>) -> Outcome {
+            let instructions = match self.usecase {
+                Usecase::CreateAccount {
+                    account_id,
+                    public_key,
+                } => create_account(account_id, public_key)?,
+                Usecase::MintPublicKey {
+                    account_id,
+                    public_key,
+                } => mint_public_key(account_id, public_key)?,
+                Usecase::BurnPublicKey {
+                    account_id,
+                    public_key,
+                } => burn_public_key(account_id.as_str(), public_key.as_str())?,
+            };
+            write!(
+                writer,
+                "{}",
+                serde_json::to_string_pretty(&instructions).wrap_err("Serialization error")?
+            )
+            .wrap_err("Failed to write.")
+        }
+    }
+
+    #[derive(Subcommand, Debug, Clone)]
+    pub enum Usecase {
+        /// Generate an instruction that creates a new account.
+        CreateAccount {
+            /// A new account id.
+            #[clap(long)]
+            account_id: Option<String>,
+            /// A new account public key.
+            /// If this flag isn't provided, will generate a new key pair.
+            #[clap(long)]
+            public_key: Option<String>,
+        },
+        /// Generate an instruction that mints public key.
+        MintPublicKey {
+            /// Mint on behalf of this account.
+            #[clap(long)]
+            account_id: Option<String>,
+            /// Mint this public key. If this flag isn't provided, will generate a new key pair.
+            #[clap(long)]
+            public_key: Option<String>,
+        },
+        /// Generate an instruction that burns the given public key.
+        BurnPublicKey {
+            /// Burn on behalf of this account.
+            #[clap(long)]
+            account_id: String,
+            /// Burn this public key.
+            #[clap(long)]
+            public_key: String,
+        },
```

Comment:
Why other instructions support omitting parameters while this one doesn't?

Reply:
In my opinion, we must know precisely what we will burn here. For example, when creating an account, we sometimes want to generate a new key pair, and we don't care what `account_id` uses for that.